npm/src/clientfilemetadataupload/index.ts

```diff
@@ -50,29 +45,37 @@ export class ClientFileMetadataUpload {
 
   async saveClientFileMetadata(
     consignmentId: string,
-    metadata: IFileMetadata[]
+    allFileMetadata: IFileMetadata[]
   ): Promise<ITdrFile[]> {
-    const clientFileData: IClientFileData[] = metadata.map((m, matchId) => {
-      const { checksum, path, lastModified, file } = m
-      //Files uploaded with 'drag and files' have '/'  prepended, those uploaded with 'browse' don't
-      //Ensure file paths stored in database are consistent
-      const validatedPath = path.startsWith("/") ? path.substring(1) : path
-      const metadataInput: ClientSideMetadataInput = {
-        originalPath: validatedPath,
-        checksum,
-        lastModified: lastModified.getTime(),
-        fileSize: file.size,
-        matchId
-      }
-      return { metadataInput, tdrFile: file }
-    })
-
-    const metadataInputs: ClientSideMetadataInput[] = clientFileData.map(
-      (cf) => cf.metadataInput
-    )
+    const {
+      metadataInputs,
+      matchFileMap
+    }: {
+      metadataInputs: ClientSideMetadataInput[]
+      matchFileMap: Map<number, File>
+    } = allFileMetadata.reduce(
+      (metadataInputsAndFileMapper, metadata: IFileMetadata, matchId) => {
```

Comment:
Minor point, but the name "mapper" confused me the first time I read it, because it makes it sound like a function rather than a value.

You could just call it something like `result`, since it's only used in the context of this `reduce`.

Reply:
Ah, ok. I was was trying to come up with something descriptive because I thought it might be confusing for someone reading it

Reply:
Yeah, it's a tough one to name because it combines two outputs. And `reduce` functions are often hard to read if you're not fluent in functional programming, so I think calling it `result` makes it clearer what role it's playing in the `reduce` function.

You have to call `result.metadataInputs` and `result.matchFileMap` to interact with it, which makes it a bit easier to follow what's going on anyway.